piptools/scripts/compile.py

```diff
@@ -378,6 +378,13 @@ def cli(
         if isinstance(output_file, LazyFile):  # pragma: no cover
             ctx.call_on_close(safecall(output_file.close_intelligently))
 
+    if resolver_name == "legacy":
+        log.warning(
```

Comment:
Why not using Deprecation warnings from warning module? Somehow I find it better.

Reply:
https://docs.python.org/3/library/exceptions.html#DeprecationWarning

> DeprecationWarning
> Base class for warnings about deprecated features when those warnings are intended for _other Python developers_.

`DeprecaetionWarning` is intended for developers, but pip-tools prints warning for end-users. Also, it is "Ignored by the default warning filters", so it should be `Warning` or something else.

<img width="681" alt="image" src="https://user-images.githubusercontent.com/7377671/201499747-3245f94e-249f-438d-8f15-968c8bd02710.png">

vs

<img width="680" alt="image" src="https://user-images.githubusercontent.com/7377671/201499667-913ea8c2-32d7-41b2-aba9-f5542750a554.png">


Personally, I prefer the latter.

Reply:
But I don't mind using the warning module if you feel that's better.

Reply:
I +1 because it was only an observation. I am ok with both approaches. The real benefit of warnings is that they are very easy to silence from outside the application.